src/components/view/ExternalAccountsView/ExternalAccountsView.tsx

```diff
@@ -1,42 +1,64 @@
-	excludeDeviceManufacturers?: boolean,
-	presentation?: ViewPresentationType,
-	preview?: boolean
+    excludeProviders?: boolean;
+    excludeHealthPlans?: boolean;
+    excludeDeviceManufacturers?: boolean;
+    presentation?: ViewPresentationType;
+    preview?: boolean;
 }
 
 export type ViewPresentationType = "Modal" | "Push";
 
 export default function (props: ExternalAccountsViewProps) {
 
-	let titleKey = props.excludeDeviceManufacturers ? "external-accounts-title-without-devices" : "external-accounts-title";
+    let title = '';
+    let externalAccountProviderCategories: string[] = [];
 
-	function onExternalAccountsLoaded(accounts: ExternalAccount[]) {
-		if (accounts.length === 0) {
-			if (props.presentation === "Modal") {
-				MyDataHelps.dismiss()
-			} else {
-				MyDataHelps.back();
-			}
-		}
-	}
+    if (!props.excludeProviders) {
+        externalAccountProviderCategories.push('Provider');
+        title += language['external-accounts-title-providers'];
+    }
+    if (!props.excludeHealthPlans) {
+        externalAccountProviderCategories.push('Health Plan');
+        if (title.length > 0) {
+            title += language['external-accounts-title-divider'];
+        }
+        title += language['external-accounts-title-health-plans'];
+    }
+    if (!props.excludeDeviceManufacturers) {
+        externalAccountProviderCategories.push('Device Manufacturer');
+        if (title.length > 0) {
+            title += language['external-accounts-title-divider'];
+        }
+        title += language['external-accounts-title-devices'];
+    }
 
-	return (
-		<Layout>
-			{props.presentation &&
-				<NavigationBar title={language[titleKey]}
-					showBackButton={props.presentation == "Push"}
-					showCloseButton={props.presentation == "Modal"} />
-			}
-			{!props.presentation &&
-				<StatusBarBackground color="var(--main-bg-color)" />
-			}
-			<ExternalAccountList previewState={props.preview ? "Default" : undefined} externalAccountProviderCategories={["Provider", "Health Plan"]} onExternalAccountsLoaded={onExternalAccountsLoaded} />
-		</Layout>
-	)
+    function onExternalAccountsLoaded(accounts: ExternalAccount[]) {
+        if (accounts.length === 0) {
+            if (props.presentation === "Modal") {
+                MyDataHelps.dismiss()
+            } else {
+                MyDataHelps.back();
+            }
+        }
+    }
+
+    return (
+        <Layout>
+            {props.presentation &&
+            <NavigationBar title={title}
+                           showBackButton={props.presentation == "Push"}
+                           showCloseButton={props.presentation == "Modal"}/>
+            }
+            {!props.presentation &&
+            <StatusBarBackground color="var(--main-bg-color)"/>
+            }
+            <ExternalAccountList previewState={props.preview ? "Default" : undefined} externalAccountProviderCategories={externalAccountProviderCategories} onExternalAccountsLoaded={onExternalAccountsLoaded}/>
```

Comment:
This is the only change that I think is not backward compatible.

Despite having the `excludeDeviceManufacturers`, which could be `false`, the external account list was never being passed the `Device Manufacturer` category, which I think was a bug.

Now the `Device Manufacturer` category will be passed to the external account list if `excludeDeviceManufacturers` is `false`.

Reply:
If this is bug fix, why do you say it's not "backward compatible?"  To me, saying it's not backward compatible implies that end users will have to adjust code for things to work - but in this case if they were trying to use the flag it wouldn't have been working and now it *will* start working without any changes, right?

Reply:
I was just saying that it is not backward compatible because it *will* require end users to adjust code if they want to retain the existing behavior.  That is to say, by default, it will now *include* device manufacturers, when previously it did not.

If an end user wants to have the "correct" behavior, given their current config, then nothing needs to be done.